From 20472664f75a4d533ddc09be71b14a5301124b69 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 07 Dec 2021 09:43:23 +0200
Subject: [PATCH] [SW-36216] gaudi2: sec: open stm and razwi addr

As per the updated hbw specification, stm and razwi address space
need not be protected using range registers.

Change-Id: Ic7763b8f5c5f73d2233e7b212b4e50d67f454a3a
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index a360e80..276fcf9 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -155,9 +155,9 @@
  * rr idx type : gaudi2_sec_if_hbw_rrs
  */
 enum gaudi2_sec_if_hbw_rrs {
-	HL_SEC_HBW_STM = 0,
-	HL_SEC_HBW_SPI_FLASH = 1,
-	HL_SEC_HBW_ICCM0_TO_RAZWI = 2,
+	HL_SEC_HBW_SPI_FLASH = 0,
+	HL_SEC_HBW_ICCM0_TO_PCIE_FW_SRAM = 1,
+	HL_SEC_HBW_PSOC_LBW = 2,
 	HL_SEC_HBW_SRAM_BGN = 3,
 	HL_SEC_HBW_SRAM_IMG = 4,
 	/*
@@ -166,7 +166,6 @@
 	 * LBW RR, but instead can use the HBW RRs to block
 	 * LBW Ranges.
 	 */
-	HL_SEC_HBW_PSOC_LBW = 5,
 	HL_SEC_HBW_SRAM = 6,
 	/*
 	 * SRAM END points to the first byte of HBM
@@ -671,32 +670,26 @@
 		hl_secure_hbw_rr_gen(rr_idx, min, max, 0, PSOC_INST_NUM, HL_SEC_ENGINE_PSOC);
 }
 
-#define SPI_FLASH_SIZE	0x1000000ull /* 16MB */
-#define DBG_SPACE_BASE	0x1000007FF8000000ull
-#define SPI_FLASH_BASE	(STM_FLASH_BASE_ADDR + STM_FLASH_SIZE)
-#define SPI_FLASH_END	(SPI_FLASH_BASE + SPI_FLASH_SIZE)
-#define SRAM_RSVD_BASE	0x1000008000000000ull
-#define DRAM_RSVD_END	0xFF00000000000000ull
+#define SPI_FLASH_SIZE		0x1000000ull /* 16MB */
+#define DBG_SPACE_BASE		0x1000007FF8000000ull
+#define SPI_FLASH_BASE		(STM_FLASH_BASE_ADDR + STM_FLASH_SIZE)
+#define SPI_FLASH_END		(SPI_FLASH_BASE + SPI_FLASH_SIZE)
+#define PCIE_FW_SRAM_RSVD_END	0x1000007FF7FFF000ull
+#define SRAM_RSVD_BASE		0x1000008000000000ull
+#define DRAM_RSVD_END		0xFF00000000000000ull
 
 static void hl_secure_hbw(struct hl_rr_eng_select rr_sel)
 {
 	u64 min, max;
 
 	/*
-	 * SPI FLASH END to DBG SPACE BASE
-	 * 0x1000007FF7000000 - 0x1000007FF8000000
+	 * SPI FLASH END to PCI FW SRAM END
+	 * Includes the reserved space at end of PCI FW SRAM.
+	 * 0x1000007FF7000000 - 0x1000007FF7FFF000
 	 */
 	min = SPI_FLASH_END;
-	max = CFG_BASE;
-	hl_secure_hbw_rr(HL_SEC_HBW_ICCM0_TO_RAZWI, min, max, rr_sel);
-
-	/*
-	 * Block STM memory space
-	 * 0x1000007FF4000000 - 0x1000007FF6000000
-	 */
-	min = STM_FLASH_BASE_ADDR;
-	max = STM_FLASH_BASE_ADDR + STM_FLASH_SIZE;
-	hl_secure_hbw_rr(HL_SEC_HBW_STM, min, max, rr_sel);
+	max = PCIE_FW_SRAM_RSVD_END;
+	hl_secure_hbw_rr(HL_SEC_HBW_ICCM0_TO_PCIE_FW_SRAM, min, max, rr_sel);
 
 	/*
 	 * Block Reserved space to HBM
@@ -892,8 +885,7 @@
 	/*
 	 * Open HBW RR that was closed in ppboot.
 	 */
-	hl_secure_hbw_rr(HL_SEC_HBW_ICCM0_TO_RAZWI, 0, 0, rr_sel);
-	hl_secure_hbw_rr(HL_SEC_HBW_STM, 0, 0, rr_sel);
+	hl_secure_hbw_rr(HL_SEC_HBW_ICCM0_TO_PCIE_FW_SRAM, 0, 0, rr_sel);
 	hl_secure_hbw_rr(HL_SEC_HBW_SRAM, 0, 0, rr_sel);
 	hl_secure_hbw_rr(HL_SEC_HBW_SPI_FLASH, 0, 0, rr_sel);
 	hl_secure_hbw_rr(HL_SEC_HBW_HBM, 0, 0, rr_sel);
